microscope_info.py

```diff
@@ -0,0 +1,66 @@
+import json
+import requests
+from custom_exceptions import OpenApiError
+from dotenv import load_dotenv
+import os
+
+load_dotenv()
+
+methods = [
+    "getVersion",       # Returns microscope version
+    "getFieldOfView",   # Returns FoV as an int value (μm)
+    "getZoomDirect",    # Returns current zoom as string hex value
+    "getFocusDirect",   # Returns current focus as string hex value
+    "getContrast",      # Returns position of the Contrast slider
+    "getSaturation",    # Returns position of the Saturation slider
+    "getSharpness"      # Returns position of the Sharpness slider
+]
+
+MICROSCOPE_URL = os.getenv("MICROSCOPE_URL")
+params = {"id": 6969}
+headers = {'Content-Type': 'application/json'}
+
+def post_request(MICROSCOPE_URL, method, params, headers):
+    url = f"{MICROSCOPE_URL}?jsonrpc=2.0&method={method}&id={params['id']}"
+
+    data = json.dumps({
+        "jsonrpc": "2.0",
+        "method": method,
+        "id": params['id'],
+    })
+
+    try:
+        resp = requests.post(url, data=data, headers=headers)
+        return resp.json()
+    except OpenApiError as e:
+        print(f"Request Error: {e}")
+
+def is_hex(s):
+    try:
+        int(s, 16)
+        return True
+    except ValueError:
+        return False
+
+def get_microscope_configuration():
+    config = {}
+    for method in methods:
```

Comment:
I would suggest using a `try... except` inside or outside this loop depending on how you want to handle a request failure. Should a single failure stop the whole process or simply place a `None` value in the `config`? In case the whole process needs to be stopped, then we should raise an `OpenApiError` from the catched exception.

Reply:
Done